flytekit/annotated/base_task.py

```diff
@@ -33,6 +35,57 @@ def kwtypes(**kwargs) -> Dict[str, Type]:
     return d
 
 
+@dataclass
+class TaskMetadata(object):
+    """
+    Create Metadata to be associated with this Task
+
+    Args:
+      cache: Boolean that indicates if caching should be enabled
+      cache_version: Version string to be used for the cached value
+      interruptible: Boolean that indicates that this task is of for interruptions and can be scheduled on nodes
+                           with lower QoS guarantees. This will directly reduce the `$`/`execution cost` associated,
+                            at the cost of performance penalties due to potential interruptions
+      deprecated: This should be used to indicate that the task is deprecated
+      retries: for retries=n; n > 0, on failures of this task, the task will be retried at-least n number of times.
+      timeout: the max amount of time for which one execution of this task should be executed for. If the execution
+                     will be terminated if the runtime exceeds the given timeout (approximately)
+     """
+
+    cache: bool = False
+    cache_version: str = ""
+    interruptible: bool = False
+    deprecated: str = ""
+    retries: int = 0
+    timeout: Optional[Union[datetime.timedelta, int]] = None
+
+    def __post_init__(self):
+        if self.timeout:
```

Comment:
can you also add a check here if cache is specified but cache_version isn't (and vice-versa)?

Reply:
good idea